code/datums/components/_component.dm

```diff
@@ -137,24 +144,19 @@
 				lookup[sig] -= src
 
 	signal_procs[target] -= sig_type_or_types
-	if(length(signal_procs[target]))
+	if(!signal_procs[target].len)
 		signal_procs -= target
 
 /datum/component/proc/InheritComponent(datum/component/C, i_am_original)
 	return
 
-/datum/component/proc/OnTransfer(datum/new_parent)
+/datum/component/proc/PreTransfer()
+	return
+
+/datum/component/proc/PostTransfer()
 	return
 
 /datum/component/proc/_GetInverseTypeList(our_type = type)
-	#if DM_VERSION >= 513
-	#warn 512 is definitely stable now, remove the old code
-	#endif
-
-	#if DM_VERSION < 512
-	//remove this when we use 512 full time
-	set invisibility = 101
```

Comment:
hard to believe that not having this in 511 meant that the entire component system literally collapsed in on itself. So strange.